### ISSUE TITLE
✨ Should Cotainer.build method accepts DirectoryID(string type)?

### ISSUE THREAD

**Issue by bun913** [dagger/dagger#4631]
### What are you trying to do?

- `Container.build` method can accept context(`Directory` type or `DirectoryID` type)
- Below is my VSCode suggestion image.

<img width="982" alt="スクリーンショット 2023-02-22 11 48 00" src="https://user-images.githubusercontent.com/73948280/220508587-548a03a5-e31e-4deb-a976-827c8f366690.png">

- When I see this suggestion firstly, I think that I can pass Docker Build Context Path(path to docker build) as argument.
- Actually, the string means [DirectoryID type](https://github.com/dagger/dagger/blob/36a3929f291bc03e2f48fd2687e5538a063c63ea/sdk/nodejs/api/client.gen.ts#L281).
- Therefor, I got error like below
- I thought it would be easier for the developer to understand if only the Directory type was accepted in a variable called context.

```json
[
	{
		"message": "failed to decode *core.directoryIDPayload bytes: illegal base64 data at input byte 15",
		"locations": [
			{
				"line": 5,
				"column": 9
			}
		],
		"path": [
			"container",
			"build"
		]
	}
]
```

### Why is this important to you?

- I don't really understand the need to accept the type DirectoryID
- Since there is a value object called Directory, wouldn't it be easier for developers to understand if it were unified with this object?
- Or I thought it would be better if the DirectoryID type could be changed to be written as a string so that it is not interpreted by the IDE as a string.
    - Sorry I'm not very familiar with TypeScript.

If you have good idea, I can challenge this ploblem.

**Comment by helderco** [comment]
Hi @bun913, can you confirm what version of the NodeJS SDK you're using? What you're describing has been fixed since [v0.3.2](https://github.com/dagger/dagger/releases/tag/sdk%2Fnodejs%2Fv0.3.2).

- https://github.com/dagger/dagger/pull/4313

**Comment by bun913** [comment]
@helderco 

Sorry, I used v0.3.0.

I've confirmed this problem is fixed.
